### PR TITLE
All components are now consistently implemented by a datatype

### PR DESCRIPTION
**Pull request by Wumpf**
### What

Previously, we were a bit inconsistent with this: Most components were backed up by a single datatype but some weren't. This is now enforced everywhere.

The expectation was that this saves a lot of code since we can forward to fewer serialized types. In reality this isn't _as_ good as expected since a bunch of new datatypes had to be introduced:

* `Float64`
* `Blob` used by `Blob` component - there should probably be no blob component but instead components with a deeper semantic
* `Utf8List` for the sole purpose of implementing `VisualizerOverrides`
* `ViewCoordinates` for the sole purpose of implementing `ViewCoordinates`

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6823?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6823?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/6823)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

**Comment by Wumpf** [comment]
@rerun-bot full-check

**Comment by Wumpf** [comment]
@rerun-bot full-check

File: crates/re_blueprint_tree/src/blueprint_tree.rs [hunk 1]
```diff
@@ -546,7 +546,7 @@ impl BlueprintTree {
                             .save_recursive_override_or_clear_if_redundant(
                                 ctx,
                                 &query_result.tree,
-                                &Visible(visible_after),
+                                &Visible(visible_after.into()),
```
Comment (emilk) [inline crates/re_blueprint_tree/src/blueprint_tree.rs:549]: I think this reads much better:

```suggestion
                                &Visible::from(visible_after),
```

here and everywhere

Comment (emilk) [reply crates/re_blueprint_tree/src/blueprint_tree.rs:549]: Or 

```suggestion
                                &Visible::new(visible_after),
```

File: crates/re_types_core/src/datatypes/float32_ext.rs [hunk 2]
```diff
@@ -1,5 +1,11 @@
 use super::Float32;
 
+impl std::fmt::Display for Float32 {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:.prec$}", self.0, prec = crate::DISPLAY_PRECISION)
```
Comment (emilk) [inline crates/re_types_core/src/datatypes/float32_ext.rs:5]: Let not force a precision if the user has picked one already

```suggestion
        let precision = f.precision().unwrap_or(crate::DEFAULT_DISPLAY_DECIMALS);
        write!(f, "{:.prec$}", self.0, prec = precision)
```

Here and everywhere.


Also: `.$` picks the number of _decimals_, not the precision (`12.345` has 3 decimals, and a precision of 5 digits)

File: crates/re_types_core/src/lib.rs [hunk 3]
```diff
@@ -20,6 +20,9 @@
 
 // ---
 
+/// Number of decimals shown for all float display methods.
+pub const DISPLAY_PRECISION: usize = 3;
```
Comment (emilk) [inline crates/re_types_core/src/lib.rs:24]:
```suggestion
pub const DEFAULT_DISPLAY_DECIMALS: usize = 3;
```

with the other changes

**Review by emilk — APPROVED** [review]
Good cleanup, though the extra layers of indirection is rather painful, with all the extra `*`, `.0` and `.into()` 😞

File: crates/re_types_builder/src/report.rs [hunk 4]
```diff
@@ -76,12 +76,12 @@ impl Report {
...
 
         while let Ok(warn) = self.warnings.try_recv() {
-            eprintln!("Warning: {warn}");
+            re_log::warn!("{warn}");
         }
 
         while let Ok(err) = self.errors.try_recv() {
             errored = true;
-            eprintln!("Error: {err}");
+            re_log::error!("{err}");
```
Comment (emilk) [inline crates/re_types_builder/src/report.rs:84]: I don't like this change.

It makes the line longer, but to what benefit? The time stamp is not useful, nor is the crate name.

If you want colored text, you can get it with other means.

File: crates/re_types_core/src/datatypes/float32_ext.rs (hunk 2 above)
Comment (Wumpf) [reply crates/re_types_core/src/datatypes/float32_ext.rs:5]: as if I wrote any original code in this PR ;). That's c&p from Vec & friends.
Fixing it in more places then

File: crates/re_blueprint_tree/src/blueprint_tree.rs (hunk 1 above)
Comment (Wumpf) [reply crates/re_blueprint_tree/src/blueprint_tree.rs:549]: We generally don't have `new` methods for components today and I don't want to make this PR _even_ more complex. `From` it is then.

File: crates/re_types_core/src/datatypes/float32_ext.rs (hunk 2 above)
Comment (Wumpf) [reply crates/re_types_core/src/datatypes/float32_ext.rs:5]: (very good change though! 👍 )

**Pull request merged**